packages/react-core/src/components/Touchspin/Touchspin.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Touchspin/touchspin';
+import { css } from '@patternfly/react-styles';
+import MinusIcon from '@patternfly/react-icons/dist/js/icons/minus-icon';
+import PlusIcon from '@patternfly/react-icons/dist/js/icons/plus-icon';
+import { Button, ButtonProps } from '../Button';
+
+export interface TouchspinProps extends React.HTMLProps<HTMLDivElement> {
+  /** Value of the touchspin */
+  value?: number;
+  /** Additional classes added to the Skeleton */
+  className?: string;
+  /** Sets the width of the touchspin to a number of characters */
+  widthChars?: number;
+  /** Indicates the whole touchspin should be disabled */
+  isDisabled?: boolean;
+  /** Callback for the minus button */
+  onMinus?: (name: string, event: React.MouseEvent) => void;
+  /** Callback for the text input changing */
+  onChange?: (event: React.FormEvent<HTMLInputElement>) => void;
+  /** Callback for the plus button */
+  onPlus?: (name: string, event: React.MouseEvent) => void;
+  /** Adds the given touchspin unit to the touchspin */
+  unit?: React.ReactNode;
+  /** Position of the touchspin unit in relation to the touchspin */
+  unitPosition?: 'before' | 'after';
+  /** Additional properties added to the touchspin text input */
+  inputProps?: any;
+  /** Additional properties added to the minus button */
+  minusButtonProps?: ButtonProps;
+  /** Additional properties added to the plus button */
+  plusButtonProps?: ButtonProps;
+}
+
+export const Touchspin: React.FunctionComponent<TouchspinProps> = ({
+  value = 0,
+  className,
+  widthChars,
+  isDisabled = false,
+  onMinus,
+  onChange,
+  onPlus,
+  unit,
+  unitPosition = 'after',
+  inputProps,
+  minusButtonProps,
```

Comment:
I am not sure about the aria-label being part of this prop.  @jessiehuff shouldn't we default the aria labels for the buttons to a useful string like we do elsewhere for some buttons (e.g. a close button)?
@kmcfaul are there other props that might be passed to the plus or minus buttons?

Reply:
I can add defaults for the aria-labels, but I wanted to allow a user to pass them in directly if they wanted custom strings or localized strings. 

You can pass any props on our Button component in with these, as well as general props like `id`.

Reply:
Also used for the thresholds to pass in the Button `isDisabled`

Reply:
So, I wonder if we want to have explicit props for these (aria-label, isDisabled, etc). Especially if it is a common use case ti disable one of the buttons.  I don't think we do these anywhere else where we let the consumer spread props directly to button in the component.
@mcarrano what do you think about this?  Should we expose an interface for the consumer to be able to disable the plus or minus button?

Reply:
I'm not sure @tlabaj . But I will say that I expected that instead of the consumer needing to write code to do limit checking and disable/enable buttons, I would have expected to just pass a min or max value and have the component handle this internally. That said, I'll defer to you about the right approach.

Reply:
I agree with `minThreshold` and `maxThreshold` properties (and will add them), but would we still want a `props` property or restrict the user to only passing in `aria-label` for each button?

Reply:
I think it would be good to add a default like you guys mention, but I don't see a significant issue with passing the `aria-label` this way. 🙂

Reply:
And `id`

Reply:
Actually `aria-label` should have a default already (`Minus` and `Plus`), so that should be good to go.

Reply:
In my opinion I think that this deviates from how we usually add aria attributes.  I also think that is is not as intuitive for the consumer.  If they don't look at the example, they would nit know how to add the props.

Reply:
Ok, I can remove the `props` object and only let the user pass in `minusButtonAriaLabel`, `plusButtonAriaLabel`, `minusButtonId`, and `plusButtonId`, if we're sure those are the only things we want the user to customize on our Button component. What about the `inputProps`?

Reply:
`props` just allows greater flexibility

Reply:
@mcarrano how flexible do we want the plus and minus buttons to be?  
@kmcfaul I don't think we typically add additional props to sub components like buttons in a component.  If we have done that in the past it was most likely an ask form a consumer.  I also think we want a prop for the input's aria-label.
this is just my opinion.

Reply:
Added dedicated `aria-label` properties for the buttons and input. Kept the additional props objects in, but they generally shouldn't be used unless the user wants something specific out of them (like adding IDs for testing).